source/components/composing-components.md

```diff
@@ -0,0 +1,202 @@
+
+```app/templates/components/user-list.hbs
+{{#if hasBlock}}
+  {{yield}}
+{{else}}
+  <p>No Block Specified</p>
+{{/if}}
+```
+
+Once we know that the component is being used in block form, we can yeild
+whatever the user has placed inside the block wherever we want via the `{{yield}}` helper.
+
+This helper can be used once, or many times. You can make your component into a type of `<ul>` element,
+that is a list that will repeat n times. Like the following example, where we can output a custom summary.
+
+```app/templates/components/post-list.hbs
+{{#each posts as |post|}}
+  <h3>{{post.title}}</h3>
+  <p>{{yield}}</p>
+{{/each}}
+```
+
+Which can be used like so:
+
+```app/templates/posts.hbs
+{{#post-list posts=headlinePosts}}
+  Greatest post ever!
+{{/post-list}}
+```
+
+And will result in the following HTML:
+
+```html
+<div id="ember123" class="ember-view">
+  <h3>Tomster goes to town</h3>
+  <p>Greatest post ever!</p>
+  <h3>Tomster on vacation</h3>
+  <p>Greatest post ever!</p>
+</div>
+```
+
+But what use is it to just output the same thing over and over? Don't we want to customize our posts,
+and display the right content? Sure we do. Lets explore the `{{yield}}` helper a bit.
+
+### Data Down
+
+To accomplish composability beyond just simple templates, we need to pass context to those templates.
+This can be done with the `{{yield}}` helper that was just introduced above.
+
+The `{{yield}}` defines where the content we defined inside our component block will yield in the component's layout,
+as we saw in the previous section. Apart from that, the yield helper also allows us to send data down, providing a context for
+the content inside of the component's block. This allows anyone using our component to get context about what's going on
+inside, and to customize the output of the component with that information.
+
+```app/templates/components/my-component.hbs
+{{yield}}
+{{yield "post"}}
+{{yield item}}
+{{yield this "footer"}}
+{{yield attrs.onchange}}
+```
+
+By default yield does not send any context, but you can provide an arbitrary number of arguments.
+Sending data down allows the consumer of our component to utilize that data.
+For the consumer to get access to the data, she needs to know what data a component exposes,
+this is where documentation is so crucial, and from there the yielded data can be accessed
+with the `as` operator. Let's take `{{yield user "Dummy Text"}}` as an example:
+
+```app/templates/users.hbs
+{{#user-list users=model as |user text|}}
+  <h3>{{text}}</h3>
+  {{user-card user=user}}
+{{/user-list}}
+```
+
+Now `{{user-card}}` has access to the current user, which changes as the `{{user-list}}` iterates over each user
+since the `{{yield user "Dummy Text"}}` is inside the `{{each}}` block.
+This opens up the possibility to use the `{{component}}` helper, for example:
+
+```app/templates/components/user-profile.hbs
+<h3>Profile</h3>
+{{yield "user-avatar" user}}
+{{yield "user-details" user}}
```

Comment:
I got confused by the multiple `{{yield}}`s at first. Did I miss the explanation?